test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -297,25 +291,64 @@ public boolean shouldCache(Query query) {
             return null;
         });
 
-        NestedDocuments nestedDocuments = new NestedDocuments(mapperService, searchContext.bitsetFilterCache()::getBitSetProducer);
-        when(searchContext.getNestedDocuments())
-            .thenReturn(nestedDocuments);
+        MultiBucketConsumer consumer = new MultiBucketConsumer(maxBucket, breakerService.getBreaker(CircuitBreaker.REQUEST));
+        BitsetFilterCache bitsetFilterCache = new BitsetFilterCache(indexSettings, mock(Listener.class));
+        return new ProductionAggregationContext(
+            queryShardContext,
+            query,
+            null,
+            consumer,
+            () -> buildSubSearchContext(mapperService, queryShardContext, bitsetFilterCache),
+            releasables::add,
+            bitsetFilterCache,
+            randomInt(),
+            () -> 0L,
+            () -> false
+        );
```

Comment:
I'd love to use a specialized testing subclass but this change is large enough as it is without me changing it.